tools/autograd/templates/Functions.cpp

```diff
@@ -1510,6 +1510,41 @@ Tensor svd_backward(const std::vector<torch::autograd::Variable> &grads, const T
   return u_term + sigma_term + v_term;
 }
 
+// http://eprints.maths.ox.ac.uk/1079/1/NA-08-01.pdf
+Tensor symeig_backward(const std::vector<torch::autograd::Variable> &grads, const Tensor& self,
+                    bool eigenvectors, bool upper, const Tensor& lambda, const Tensor& v) {
+    auto n = lambda.size(0);
+    auto glambda = grads[0];
+    auto gv = grads[1];
+    
+    auto vt = v.t();
+    
+    if (!eigenvectors) {
+        throw std::runtime_error(std::string("cannot compute backward without "
+                                             "computing eigenvectors in forward pass"));
+    }
+    
+    Tensor result = at::zeros_like(self);
```

Comment:
nit: you can just do `Tensor result;` here and assign zeros in the else of `if (gv.defined())`